client/landing/stepper/declarative-flow/internals/steps-repository/site-migration-upgrade-plan/index.tsx

```diff
@@ -18,26 +17,19 @@ import { useQuery } from 'calypso/landing/stepper/hooks/use-query';
 import { useSite } from 'calypso/landing/stepper/hooks/use-site';
 import { useSiteSlug } from 'calypso/landing/stepper/hooks/use-site-slug';
 import { recordTracksEvent } from 'calypso/lib/analytics/tracks';
-import type { StepProps } from '../../types';
-import './style.scss';
-
-type StepContainerProps = React.ComponentProps< typeof StepContainer >;
+import type { Step } from '../../types';
 
-interface Props extends StepProps {
-	skipLabelText?: StepContainerProps[ 'skipLabelText' ];
-	onSkip?: StepContainerProps[ 'goNext' ];
-	skipPosition?: StepContainerProps[ 'skipButtonAlign' ];
-	headerText?: string;
-	customizedActionButtons?: StepContainerProps[ 'customizedActionButtons' ];
-}
+import './style.scss';
 
-const SiteMigrationUpgradePlan: FC< Props > = ( {
-	navigation,
-	data,
-	customizedActionButtons,
-	flow,
-	...props
-} ) => {
+const SiteMigrationUpgradePlan: Step< {
+	accepts: {
+		skipLabelText?: string;
+		onSkip?: () => void;
+		skipPosition?: 'top' | 'bottom';
+		headerText?: string;
+		customizedActionButtons?: React.ReactElement;
+	};
+} > = ( { navigation, data, customizedActionButtons, flow, ...props } ) => {
```

Comment:
Maybe we can add submit types here

Reply:
For reference, it looks like the union of all types would be:
```ts
{
	goToCheckout?: boolean;
	plan?: string; // This may be a plan path slug for checkout, and not the plan slug itself -- see L67
	sendIntentWhenCreatingTrial?: boolean; // this may be deprecated/disabled, but the code is still there
	verifyEmail?: boolean;
}
```

Maybe this makes sense as a dedicated follow-on task to get this explicitly typed?

Reply:
Oh this uncovered that you if you type `accepted` you can get away with not typing `submits`, checking..

Reply:
Fixed